lib/buffer.js

```diff
@@ -134,7 +134,11 @@ function fromObject(obj) {
     return b;
   }
 
-  if (Array.isArray(obj)) {
+  if (obj == null) {
+    throw new TypeError('Must start with number, buffer, array or string');
+  }
+
+  function fromArrayLike(obj) {
```

Comment:
I wonder if it might be beneficial (performance-wise) to move this function out of `fromObject()` since it does not depend on any outside variables.